.github/workflows/release.yml

```diff
@@ -0,0 +1,93 @@
+name: Release
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag name to release'
+        required: true
+
+jobs:
+
+  release:
+    name: Create Release From Tag
+    runs-on: ubuntu-latest
+    steps:
+      - name: Initialize tag via push
+        if: github.event_name == 'push'
+        run: echo "VER_TAG=${GITHUB_REF/refs\/tags\//}" >> $GITHUB_ENV
+
+      - name: Initialize tag via manual dispatch
+        if: github.event_name == 'workflow_dispatch'
+        run: echo "VER_TAG=${{ github.event.inputs.tag }}" >> $GITHUB_ENV
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Build Release
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install setuptools wheel twine
+
+          python setup.py sdist bdist_wheel
+
+          # expected_asset=fh_immuta_utils-${VER_TAG}.tar.gz
+          # for testing, we will use fake version numbers that don't match the built package
+          # so we're hardcoding this for testing purposes
+          expected_asset=fh_immuta_utils-0.2.0.tar.gz
+
+          if [ -e "dist/${expected_asset}" ]
+          then
+            echo "RELEASE_ASSET=${expected_asset}" >> $GITHUB_ENV
+          else
+            exit 1
+          fi
+
+      - name: Test Wheel
+        run: python -m twine check dist/*
+
+      - name: Release Notes
+        run: |
+          expected="release_notes/v${VER_TAG}.md"
+          if [ -e "$expected" ]
+          then
+            echo "BODY_PATH=$expected" >> $GITHUB_ENV
+          else
+            generated=gen/v${VER_TAG}.md
+            echo "Release version ${VER_TAG}" > $generated
+            echo "BODY_PATH=$generated" >> $GITHUB_ENV
+          fi
+
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ env.VER_TAG }}
+          release_name: v${{ env.VER_TAG }}
```

Comment:
nit request to remove the preceding `v` to keep with the current pattern for releases